src/libtiled/wangset.cpp

```diff
@@ -288,11 +473,62 @@ bool WangSet::wangIdIsValid(WangId wangId) const
         if (wangId.edgeColor(i) > mEdgeColors
                 || wangId.cornerColor(i) > mCornerColors)
             return false;
+
+        if (mEdgeColors <= 1)
+            if (wangId.edgeColor(i))
+                return false;
+
+        if (mCornerColors <= 1)
+            if (wangId.cornerColor(i))
+                return false;
     }
 
     return true;
 }
 
+bool WangSet::wangIdIsUsed(WangId wangId) const
+{
+    unsigned mask = 0;
+
+    if (mEdgeColors > 1)
+        mask |= 0x0f0f0f0f;
+    if (mCornerColors > 1)
+        mask |= 0xf0f0f0f0;
+
+    for (WangId usedId : mWangIdToWangTile.keys()) {
```

Comment:
That is originally what I was doing, but ran into some problems when there was less certainty about the wangId (I was searching for something with at least all ones in a corner only set, but the wangSet had the corners as wildcards (0)). The mask just allows that, though it should now never happen, the spaces in the wangId which don't matter can be different. Though now that I have made more clear rules about wangIds, this could be changed.